autorest/codegen/__init__.py

```diff
@@ -76,6 +76,21 @@ def _validate_code_model_options(options: Dict[str, Any]) -> None:
             "If you want operation files, pass in flag --show-operations"
         )
 
+    if options["package_mode"] == 'dataplane' and \
+        not all([options["package_name"], options["client_name"], options["credential_scopes"],
+                options["package_pprint_name"], options["output_folder"], options["input_file"]]):
+        raise ValueError(
+            "--package-name, --title, --package-pprint-name, --credential-scopes, --output-folder"
```

Comment:
I don't think all of these flags need to be required for package-mode to work. For example, at least for title, we can get the value from the client name

Reply:
Indeed, most of those flags will be required anyway, and we should get the values from the codemodel, once they are solved and resolved. For instance, the codemodel will always have a title.

For options that are not necessary today to generate, see that I did a proposal in the initial issue, and it was not to add new flags.

Reply:
Hi @lmazuel @iscai-msft, when people want to generate a shippable package, these flags(package_name, output_folder, etc) are necessary in [swagger/readme.md](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/webpubsub/azure-messaging-webpubsubservice/swagger#settings). Rather than that let users configure readme.md then generate package with the configuration, I think it is more convenient to do that with one step: force users to input necessary flags with 'package-mode`. After generation, users could get shippable package directly. It may be a little different with https://github.com/Azure/autorest.python/issues/1053, but it shall be convenient for users, too.

`title`: although it is usually defined in swagger, but most of it is invalid for SDK like [this](https://github.com/Azure/azure-rest-api-specs/blob/679887ace44697c726aba8d2814ee415a5d25e6f/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json#L4). So I think it is better to let users input for confirmation.

I try to let users input less parameters as soon as possible, now they only need 6.

Reply:
@msyyc I'm kind of confused, wouldn't people be running `--package-mode` with generation, so there's no need to add duplicate flags explicitly for pakcage mode?

For example, `--title` shouldn't be required because that's a flag to override the client name in the swagger. Don't we want the package name's client to be the same as the generated client's name? So i don't get why you want to require `--title` for package mode. Honestly the only flag I think we may need to do a check on is `--package-pprint-name`, if we decide we do want to add this flag, the other checks should be the same for generation